controllers/watchers/target_pod.go

```diff
@@ -0,0 +1,563 @@
+	newTargetName, newTargetKind := getTargetNameAndKind(newObj)
+
+	d.Log.Debugw("DisruptionTargetHandler UPDATE",
+		"disruptionName", d.Disruption.Name,
+		"disruptionNamespace", d.Disruption.Namespace,
+		"oldTargetName", oldTargetName,
+		"oldTargetKind", oldTargetKind,
+		"newTargetName", newTargetName,
+		"newTargetKind", newTargetKind,
+	)
+
+	d.OnChangeHandleMetricsSink(newPod, newNode, okNewPod, okNewNode)
+
+	if d.EnableObserver {
+		d.OnChangeHandleNotifierSink(oldPod, newPod, oldNode, newNode, okOldPod, okNewPod, okOldNode, okNewNode)
+	}
+}
+
+// OnChangeHandleMetricsSink Trigger Metric Sink on changes in the targets
+func (d DisruptionTargetHandler) OnChangeHandleMetricsSink(pod *v1.Pod, node *v1.Node, okPod, okNode bool) {
+	switch {
+	case okPod:
+		d.handleMetricSinkError(d.MetricsSink.MetricSelectorCacheTriggered([]string{"disruptionName:" + d.Disruption.Name, "namespace:" + d.Disruption.Namespace, "event:add", "targetKind:pod", "target:" + pod.Name}))
+	case okNode:
+		d.handleMetricSinkError(d.MetricsSink.MetricSelectorCacheTriggered([]string{"disruptionName:" + d.Disruption.Name, "namespace:" + d.Disruption.Namespace, "event:add", "targetKind:node", "target:" + node.Name}))
+	default:
+		d.handleMetricSinkError(d.MetricsSink.MetricSelectorCacheTriggered([]string{"disruptionName:" + d.Disruption.Name, "namespace:" + d.Disruption.Namespace, "event:add", "targetKind:object"}))
+	}
+}
+
+// OnChangeHandleNotifierSink Trigger Notifier Sink on changes in the targets
+func (d DisruptionTargetHandler) OnChangeHandleNotifierSink(oldPod, newPod *v1.Pod, oldNode, newNode *v1.Node, okOldPod, okNewPod, okOldNode, okNewNode bool) {
+	var objectToNotify runtime.Object
+
+	eventsToSend := make(map[string]bool)
+	name := ""
+
+	switch {
+	case okNewPod && okOldPod:
+		objectToNotify, name = newPod, newPod.Name
+
+		disruptionEvents, err := d.getEventsFromCurrentDisruption("Pod", newPod.ObjectMeta, d.Disruption.CreationTimestamp.Time)
+		if err != nil {
+			d.Log.Warnf("couldn't get the list of events from the target. Might not be able to notify on error changes: %s", err.Error())
+		}
+
+		// we detect and compute the error / warning events, status changes, conditions of the updated pod
+		eventsToSend = d.buildPodEventsToSend(*oldPod, *newPod, disruptionEvents)
+	case okNewNode && okOldNode:
+		objectToNotify, name = newNode, newNode.Name
+
+		disruptionEvents, err := d.getEventsFromCurrentDisruption("Node", newNode.ObjectMeta, d.Disruption.CreationTimestamp.Time)
+		if err != nil {
+			d.Log.Warnf("couldn't get the list of events from the target. Might not be able to notify on error changes: %s", err.Error())
+		}
+
+		// we detect and compute the error / warning events, status changes, conditions of the updated node
+		eventsToSend = d.buildNodeEventsToSend(*oldNode, *newNode, disruptionEvents)
+	default:
+		d.Log.Debugw("target observer couldn't detect what type of changes happened on the targets")
```

Comment:
```suggestion
		d.Log.Warnw("target observer couldn't detect what type of changes happened on the targets")
```

At least Warn, if not Error?

Reply:
Yes, it is not normal if the default condition is true.